src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
+  }
+
+  public findDuplicates(arr: number[]): number[] {
+    const duplicates: number[] = [];
+    for (let i = 0; i < arr.length; i++) {
+      for (let j = i + 1; j < arr.length; j++) {
+        if (arr[i] === arr[j]) {
+          duplicates.push(arr[i]);
+        }
+      }
+    }
+    return duplicates;
+  }
+
+  public inefficientSort(arr: number[]): number[] {
+    return arr.sort((a, b) => {
+      if (a > b) return 1;
+      if (a < b) return -1;
+      return 0;
+    });
+  }
+
+  public sendNotification(type: 'email' | 'sms', message: string): void {
+    if (type === 'email') {
+      this.sendEmail(message);
+    } else if (type === 'sms') {
+      this.sendSms(message);
+    } else {
+      console.log('Invalid notification type');
+    }
+  }
+
+  private sendEmail(message: string): void {
+    console.log(`Sending email with message: ${message}`);
+  }
+
+  private sendSms(message: string): void {
+    console.log(`Sending SMS with message: ${message}`);
+  }
+
+  public divide(a: number, b: number): number {
+    try {
+      return a / b;
+    } catch (error) {
+      console.error('An error occurred during division:', error);
+      return 0;
+    }
+  }
+
+  public async fetchDataFromApi(apiUrl: string): Promise<string> {
+    const response = await fetch(apiUrl); // Missing proper error handling
+    return await response.text();
```

Comment:
PullProofAI

The `ExampleService` class contains multiple methods with different responsibilities. It's recommended to follow the Single Responsibility Principle and split these methods into separate classes or services based on their functionalities. This will improve code maintainability and readability.



<!-- This is an auto-generated comment by PullProofAI -->